File: 3657-check-if-grid-can-be-cut-into-sections/3657-check-if-grid-can-be-cut-into-sections.ts
function checkValidCuts(n: number, rectangles: number[][]): boolean {
    function checkCuts(intervals: [number, number][]): boolean {
        intervals.sort((a, b) => a[0] - b[0]);
        let lastEnd = intervals[0][1];
        let gapCount = 0;

        for (let interval of intervals){
            if (interval[0] >= lastEnd)
                gapCount++;

            lastEnd = Math.max(lastEnd, interval[1]);
        }

        return gapCount >= 2;
    }

    const xIntervals: [number, number][] = [];
    const yIntervals: [number, number][] = [];

    for (let rect of rectangles){
        xIntervals.push([rect[0], rect[2]]);
        yIntervals.push([rect[1], rect[3]]);
    }

    return checkCuts(xIntervals) || checkCuts(yIntervals);

};

File: 3657-check-if-grid-can-be-cut-into-sections/3657-check-if-grid-can-be-cut-into-sections.cs
public class Solution {
    public bool CheckValidCuts(int n, int[][] rectangles) {
        var xIntervals = rectangles.Select(rect => new int[]{
            rect[0], rect[2]
        }).ToArray();

        var yIntervals = rectangles.Select(rect => new int[]{
            rect[1], rect[3]
        }).ToArray();

        return Check(xIntervals) || Check(yIntervals); 
    }

    private bool Check(int[][] intervals){
        Array.Sort(intervals, (a, b) => a[0].CompareTo(b[0]));

        int gapCount = 0;
        int lastEnd = intervals[0][1];

        foreach (var interval in intervals){
            if (interval[0] >= lastEnd)
                gapCount++;

            lastEnd = Math.Max(lastEnd, interval[1]);
        }

        return gapCount >= 2;
    }
}